addons/website_profile/controllers/main.py

```diff
@@ -84,14 +88,16 @@ def get_user_profile_avatar(self, user_id, field='avatar_256', width=0, height=0
             field_name=field, width=int(width), height=int(height), crop=crop
         ).get_response()
 
-    @http.route(['/profile/user/<int:user_id>'], type='http', auth="public", website=True)
+    @http.route(['/profile/user/<int:user_id>'], type='http', auth='user', website=True)
     def view_user_profile(self, user_id, **post):
-        user = self._check_user_profile_access(user_id)
-        if not user:
-            return request.render("website_profile.private_profile")
+        access = self._check_user_profile_access(user_id)
+        if access['not_found']:
+            raise request.not_found()
+        if access['denial']:
+            return request.render('website_profile.profile_access_denied', {'reason': access['denial']})
```

Comment:
`denial_reason` could be a value to use both in the controller methods and in the template(s).